File: backend/dao/ProcessingObject.py
from enum import IntEnum

class p_type(IntEnum):
    DOWNLOADING = 1
    ANALYSIS = 2

class p_status(IntEnum):
    WAITTING = 1
    PROCESSING = 2
    CANCELED = 3
    DONE = 4
    
class ProcessingObject:
    def __init__(self, _project_path, _ptype):
        self.ptype = _ptype
        self.project_path = _project_path
        self.progress = 0.0
        self.status = p_status.WAITTING
    def to_dict(self):
        return {
            'ptype' : self.ptype,
            'project_path' : self.project_path,
            'progress' : self.progress,
            'status' : self.status
        }

File: src/objects/ProcessingObject.ts
enum ptype {
    DOWNLOADING=1,
    ANALYSIS,

}
enum pstatus {
    WAITTING=1,
    PROCESSING,
    CANCELED,
    DONE
}
class ProcessingObject {
    ptype: ptype
    project_id: string
    progress: number
    status:pstatus
    info: string
    project_front_end_id: string
    project_path: string
    constructor(project_id:string,ptype: ptype, project_front_end_id, project_path) {
        this.ptype = ptype
        this.project_id = project_id
        this.progress = 0.0
        this.status = pstatus.WAITTING
        if (project_front_end_id == '') {
            // 丢出警报
        } else {
            this.project_path = project_path
        }
    }

}
export { ProcessingObject, pstatus, ptype }